### PR TITLE
regex for userlogin and simple factory

### PR DESCRIPTION
**Pull request by alanchen34**

### CODE REVIEW

File: src/main/java/Controller/LoginController.java [hunk 1]
```diff
@@ -18,12 +18,22 @@ public LoginController(){
      */
     public LoginController(UserManager userManager){this.uman = userManager;}
 
+    /**
+     * check if the username is a valid name for register
+     * @param name username
+     *
+     */
+    public boolean is_valid_name(String name){
+        return name.matches("^[a-zA-Z0-9]{4,12}$");
```
Comment (Waannng) [inline src/main/java/Controller/LoginController.java:27]: It's better to put specific process in UserManager (use case layer).

File: src/test/java/Controller/LoginControllerTest.java [hunk 2]
```diff
@@ -85,4 +93,21 @@ public void getUman() {
...
+        String d = "abcd12bc";
+        String e = "asrae123!";
+        String f = "123456789abc3";
+        assertTrue(log.is_valid_name(b));
+        assertTrue(log.is_valid_name(c));
+        assertTrue(log.is_valid_name(d));
+        assertFalse(log.is_valid_name(a));
+        assertFalse(log.is_valid_name(e));
+        assertFalse(log.is_valid_name(f));
+    }
```
Comment (Waannng) [inline src/test/java/Controller/LoginControllerTest.java:112]: move to UserManagerTest

**Review by Waannng — COMMENTED** [review]
.

File: src/test/java/Controller/LoginControllerTest.java (hunk 2 above)
Comment (alanchen34) [reply src/test/java/Controller/LoginControllerTest.java:112]: ok got it

File: src/main/java/Controller/LoginController.java (hunk 1 above)
Comment (alanchen34) [reply src/main/java/Controller/LoginController.java:27]: ok

**Pull request closed without merging**